src/components/ui/Suggest.jsx

```diff
@@ -50,57 +63,74 @@ const Suggest = ({
     }, SUGGEST_DEBOUNCE_WAIT);
 
     const handleInput = e => {
-      fetchItems(e.target.value);
+      const { value } = e.target;
+      if (value !== lastQuery) {
+        fetchItems(value);
+        setIsOpen(true);
+      }
```

Comment:
The value of `lastQuery` is not updated here during typing: as far as I understand the component is not re-rendered and the updated state value is not accessible.

Actually I am not sure this check is useful. `fetchItems` is already responsible for caching results, so maybe we could remove this check completely.

Reply:
I think you're right about removing this check. `lastQuery` was updated in `fetchItems`. I moved this update in this `handleInput` function, thanks. `lastQuery` is notably used to restore what the user has previously typed when cycling over suggests results with arrow keys. (after one complete cycle, the typed value is restored).